tests/resource/invalid/type/control_flow/access_match_arms_variable.mamba

```diff
@@ -0,0 +1,5 @@
+match 10
+    n =>
+        print("10")
```

Comment:
What about a test where we use an undefined variable in match arm body
Also one where we shadow a variable, but use it in the same manner as we would've the outer variable (e.g. assuming it is an `Int` while it is a `String`)

Reply:
No system for the second, though this is somewhat related to #375